<pattern_generator type="MogsSothPatternGenerator">
    <speed_up> 1.0 </speed_up>
<!--           <test_derivative> true </test_derivative> -->
    <constraint type="JointLimitConstraint"/>            
    
    <constraint type="EndEffectorConstraint">
        <body>  r_wrist </body>
        <position> 0.05 0 0.0 </position>
        <threshold> 0.00 </threshold>
        <desired type="body">
            <target_robot>Box_2cm</target_robot>
            <target_body> MainBody </target_body>                    
        </desired>
    </constraint>	
            <constraint type="OrientationConstraint">
            <body> r_wrist </body>
            <rotation> 1.57 1.57 0.0 </rotation>
            <desired type="body">
                <target_robot>Box_2cm</target_robot>
                <target_body> MainBody </target_body>                    
            </desired>
    </constraint>
    
        <constraint type="MinimizeQConstraint"/> 
</pattern_generator>
